src/rebar3_bsp_prv.erl

```diff
@@ -35,38 +35,55 @@ do(State) ->
       rebar3_bsp_connection:generate(Dir);
     false ->
       setup_name(State),
-      start_agent(State)
+      start_server(State)
   end,
   {ok, State}.
 
 -spec format_error(any()) -> iolist().
 format_error(Reason) ->
   io_lib:format("~p", [Reason]).
 
--spec start_agent(rebar_state:t()) -> no_return().
-start_agent(State) ->
-  simulate_proc_lib(),
-  true = register(?AGENT, self()),
-  {ok, GenState} = rebar3_bsp_agent:init(State),
-  gen_server:enter_loop(rebar3_bsp_agent, [], GenState, {local, ?AGENT}, hibernate).
+-spec start_server(rebar_state:t()) -> no_return().
+start_server(State) ->
```

Comment:
Seems like an improvement over the previous hack. Should we try and explain in a comment how the mechanism works (e.g. the message to `init`)?